server/src/main/java/io/druid/metadata/SQLMetadataConnector.java

```diff
@@ -254,6 +254,7 @@ public void createSegmentTable(final String tableName)
                 + "  partitioned BOOLEAN NOT NULL,\n"
                 + "  version VARCHAR(255) NOT NULL,\n"
                 + "  used BOOLEAN NOT NULL,\n"
+                + "  used_update_date VARCHAR(255),\n"
```

Comment:
would be nice to add a hook in the metadata connector to alter existing tables if the column is not present may be we can also set the default for this column as current time, so that users need not manually go and run alter table statements.

Reply:
will try and add.

Reply:
I was thinking of adding a hook at https://github.com/druid-io/druid/blob/master/server/src/main/java/io/druid/metadata/SQLMetadataConnector.java#L530 . However, I can't seem to find the standard sql statements to check if a particular column does not exist and then create it and that makes it non-trivial.

Reply:
I'm not sure that there is standard SQL for this. Also, ALTER TABLE commands to add columns can cause a lot of downtime on some databases, so it might be harsh to run anyway. So far we have never done this sort of automatic table migration. I think we haven't even required manual ones ever (except maybe in the distant past).